src/python_testing/TC_CLCTRL_5_1.py

```diff
@@ -676,6 +662,43 @@ async def test_TC_CLCTRL_5_1(self):
                 e.status, Status.Success, f"Failed to send command TestEventTrigger: {e.status}")
             pass
 
+        # STEP 12a: If the CL feature is not supported, skip steps 12b to 12d
+        self.step("12a")
+
+        if not is_cl_feature_supported:
+            logging.info("Calibration feature not supported, skipping steps 12b to 12d")
+
+            # Skipping steps 12b to 12d
+            self.mark_step_range_skipped("12b", "12d")
+        else:
+            # STEP 12b: TH sends command Calibrate to DUT
+            self.step("12b")
+
+            sub_handler.reset()
+            try:
+                await self.send_single_cmd(cmd=Clusters.ClosureControl.Commands.Calibrate(), endpoint=endpoint, timedRequestTimeoutMs=1000)
+            except InteractionModelError as e:
+                asserts.assert_equal(
+                    e.status, Status.Success, f"Failed to send command Calibrate: {e.status}")
+                pass
+
+            # STEP 12c: TH reads from the DUT the MainState attribute
+            self.step("12c")
+
+            mainstate = await self.read_clctrl_attribute_expect_success(endpoint=endpoint, attribute=attributes.MainState)
+            # Check if the MainState attribute has the expected values
+            asserts.assert_equal(mainstate, Clusters.ClosureControl.Enums.MainStateEnum.kCalibrating,
+                                 "MainState is not in the expected state")
+            logging.info(f"Mainstate: {mainstate}")
```

Comment:
Would give more info if mainstate is logged before the assert as in styep 7c. After the assert mainstate will always be Calibrating.

```suggestion
            logging.info(f"Mainstate: {mainstate}")
            asserts.assert_equal(mainstate, Clusters.ClosureControl.Enums.MainStateEnum.kCalibrating,
                                 "MainState is not in the expected state")
```